### ISSUE TITLE
[Bug]: Chinese AWS regions do not have a root user

### ISSUE THREAD

**Issue by daftkid** [toniblyx/prowler#1571]
**What happened?**
I can see that the prowler report says that I have root user access keys in the target AWS account. The problem here is the fact that Chinese AWS does not have such an object as root user. I spent some time debugging the problem and found that `root_account` column even does not exists in IAM credential report

**How to reproduce it**
1. What command are you running?
`./prowler -r cn-north-1 -M csv,html -c check112 -k`
3. Environment you have, like single account, multi-account, organizations, etc.
Single AWS account within the Chinese partition of AWS
5. See error
```
 Generating AWS IAM Credential Report... -  []
0.2 Saving IAM Credential Report ... - NOT_SCORED [SUPPORT]
       INFO! IAM Credential Report saved in /var/folders/j3/vpgktzcs41l8p2w_9ngzwf0h0000gq/T/prowler.cred_report-XXXXXX.SAzGP0o8
1.12 [check112] Ensure no root account access key exists - iam [Critical]
       FAIL! cn-north-1: Found access key 1 for root
       FAIL! cn-north-1: Found access key 2 for root

```

**Expected behavior**
Prowler should let the user know that root user is not available in the Chinese regions and skip the further check. For now, I can see neither region check nor missing `root_account` field check here https://github.com/prowler-cloud/prowler/blob/2.12.1/checks/check112

**Screenshots or Logs**
Also, you can add logs (anonymize them first!). Here a command that may help to share a log
`bash -x ./prowler -options > debug.log 2>&1` then attach here `debug.log`

**From where are you running Prowler?**
Please, complete the following information:
 - Resource: [e.g. EC2 instance, Fargate task, Docker container manually, EKS, Cloud9, CodeBuild, workstation, etc.)
 - OS: [e.g. Amazon Linux 2, Mac, Alpine, Windows, etc. ]
 - AWS-CLI Version [`aws --version`]:
 - Prowler Version [`./prowler -V`]: 2.12.0-27October2022
 - Shell and version:
 - Others:

**Additional context**
Please find the output of the credential report below:
```
user,arn,user_creation_time,password_enabled,password_last_used,password_last_changed,password_next_rotation,mfa_active,access_key_1_active,access_key_1_last_rotated,access_key_1_last_used_date,access_key_1_last_used_region,access_key_1_last_used_service,access_key_2_active,access_key_2_last_rotated,access_key_2_last_used_date,access_key_2_last_used_region,access_key_2_last_used_service,cert_1_active,cert_1_last_rotated,cert_2_active,cert_2_last_rotated
test_user,arn:aws-cn:iam::xxxxxxxxxxxx:user/test_user,2022-04-21T15:46:07+00:00,true,2022-05-30T14:21:57+00:00,2022-04-26T07:37:19+00:00,2022-07-25T07:37:19+00:00,true,true,2022-04-21T15:46:09+00:00,2022-06-01T10:13:00+00:00,cn-north-1,sts,false,N/A,N/A,N/A,N/A,false,N/A,false,N/A
```

**Comment by daftkid** [comment]
I'm going to check if there is a case for V3 of Prowler.

**Comment by daftkid** [comment]
The same is true for the Prowler v3 at some point. It ignores root access keys, however, still throws a false FAIL check for root_mfa_enabled.  Here it checks for Account MFA enabled, and as Chinese AWS does not have a root account principal, `AccountMFAEnabled` will always be `0`. (https://github.com/prowler-cloud/prowler/blob/bb5ba8c37c38986dd6e674dbc96e0d93f9d411af/prowler/providers/aws/services/iam/iam_root_hardware_mfa_enabled/iam_root_hardware_mfa_enabled.py#L14)

Please see the AWS cli output below with the details for my account:
```
 aws iam get-account-summary --region cn-north-1| grep MFA
        "AccountMFAEnabled": 0,
        "MFADevices": 4,
        "MFADevicesInUse": 4,
```

**Comment by MrCloudSec** [comment]
Thank you for the information @daftkid !
Please, can you execute `./prowler.py aws -c iam_root_hardware_mfa_enabled -f cn-north-1 --verbose` in the branch `fix-cn-partition`?

**Comment by daftkid** [comment]
Hi @sergargar . Seems to be good now

```
Executing 1 checks, please wait...


Check ID: iam_root_hardware_mfa_enabled - iam [critical]
Something went wrong in iam_root_hardware_mfa_enabled, please use --log-level ERROR
        INFO There are no resources

-> Scan completed! |▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉| 1/1 [100%] in 13.1s

 There are no findings in Account 459729599995
```

**Comment by jfagoagas** [comment]
Hi @daftkid, thanks for testing it. 

I see that some errors has been thrown underneath due to `Something went wrong in iam_root_hardware_mfa_enabled, please use --log-level ERROR`.

Can you run `./prowler.py aws -c iam_root_hardware_mfa_enabled -f cn-north-1 --log-level ERROR` and paste the output here? 

Working with the AWS-CN accounts sometimes it's difficult.

Thanks!

**Comment by daftkid** [comment]
```
Something went wrong in iam_root_hardware_mfa_enabled, please use --log-level ERROR
2022-12-21 22:05:51,448 [File: check.py:293]    [Module: check]  ERROR: iam_root_hardware_mfa_enabled -- UnboundLocalError[290]: local variable 'report' referenced before assignment
```

**Comment by jfagoagas** [comment]
> ```
> Something went wrong in iam_root_hardware_mfa_enabled, please use --log-level ERROR
> 2022-12-21 22:05:51,448 [File: check.py:293]    [Module: check]  ERROR: iam_root_hardware_mfa_enabled -- UnboundLocalError[290]: local variable 'report' referenced before assignment
> ```

Thanks!

**Comment by MrCloudSec** [comment]
Hi @daftkid, can you test it again in the branch `fix-cn-partition`, please?

**Comment by daftkid** [comment]
Hi @sergargar please see below:
```
2022-12-22 13:52:39,979 [File: iam_service.py:153]      [Module: iam_service]    ERROR: cn-north-1 -- NoSuchEntityException[139]: An error occurred (NoSuchEntity) when calling the GetAccountPasswordPolicy operation: The Password Policy with domain name <my_acc_id> cannot be found.

-> Scan completed! |▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉| 1/1 [100%] in 7.2s

 There are no findings in Account <my_acc_id>

```

**Comment by daftkid** [comment]
BTW I did some scans earlier today against a set of Chinese AWS accounts, prowler looks much better than yesterday in terms of Chinese partition. Thanks for this quick fix!

**Comment by MrCloudSec** [comment]
Thank you @daftkid !! Since the original issue is solved, I proceed to close this issue. 
The above IAM error will be solved in 3.0.1 version.

**Comment by MrCloudSec** [comment]
@daftkid here's the PR fixing the IAM Error: https://github.com/prowler-cloud/prowler/pull/1589
